lips/lip-10.md

```diff
@@ -0,0 +1,231 @@
+  * Each libra-id-domain is uniquely associated with single VASP
+
+## Multiple accounts per user
+VASPs must not assume that Libra ID uniquely identifies the user within the system. Users that have accounts with different VASPs would have different Libra ID within each VASP.
+
+Furthermore, a user can have multiple Libra IDs within the same VASP. This does not affect the protocol, but VASPs can leverage this - for example to allow users to create multiple ‘accounts’ like checking, savings, etc.
+
+VASPs can also allow users to generate temporary / situational Libra IDs. This can be used in many scenarios such as:
+
+* Temporary Libra ID for some merchant payments, when user don’t want to share their ‘main’ Libra ID with the merchant
+* Publicly shared Libra IDs for the various campaigns, like donations, etc.
+
+## Libra ID domains
+Each Libra ID domain is associated with a VASP. The source of truth for this association is a blockchain - a special LibraIdDomains resource must be published into VASP for it to expose its LibraID domain.
+
+```
+resource struct LibraIdDomains {
+    domains: vector<LibraIdDomain>,
+}
+
+struct LibraIdDomain {
+    domain: vector<u8>,     // Utf-8 encoding
+    wallet_uri: vector<u8>, // ASCII encoding (use url-escape for non-ASCII characters)
+}
+```
+
+* The `LibraIdDomains` resource can only be published into an account with `Role == PARENT_VASP_ROLE_ID`.
+* The `LibraIdDomains` contains a list of `LibraIdDomain` structs that can be associated with a VASP. As such, it is possible to register more than one Libra ID Domain for a given VASP. We are providing this opportunity to support possible situations when two VASPs merge into one, or when a company wants to provide multiple wallet apps, while having only a single VASP parent account on the chain.
+* Only special Treasury Compliance account (address `0xB1E55ED`) can manipulate LibraIdDomains resource:
+  * Only TC account can create and publish LibraIdDomains resource
+  * Only TC account can add, remove or update an LibraIdDomain within LibraIdDomains resource
+* In order to register a Libra Id domain, VASP needs to submit a request to Libra Association. LA will perform certain checks (out of scope of this document) before they will issue an on-chain transaction to register a Libra Id Domain. The checks that LA will perform will make sure that VASPs do not claim irrelevant Libra Id Domains. LA submission will also ensure that Libra Id Domains are unique.
+* It should also be possible (and probably going to be most common) for a new VASP to submit their desired Libra ID domain(s) along with the request to create a `ParentVASP` account.
+* Association may provide an open source application(*Indexer*) that will allow indexing available Libra IDs and will provide a convenient REST api for applications to fetch information about the domains. The API of such an indexer is out of scope of this RFC.
+
+# Peer-2-peer payment protocol
```

Comment:
Can we make this something like "Utilization of LibraID during payments" and not tie it so specifically to p2p payments?  I kind of think we should tweak this whole section to make it just describe how libra ID works under the hood.  The internals of it should be the same for any flow that is used. I think even for merchant flows, the APIs should end up working in a very similar way.  I would expect a merchant flow to really just layer on top of the typical payment case.  So in a merchant flow:
* Enter your libraID on the merchant's page
* some magic happens
* app-switch to your wallet app
* your wallet pre-fills the txn details (inclusive of the receiver's libra ID)
* from this point, for a general merchant payment (and not an auth, recurring, etc.), the merchant wrapper ends and the flow described in this doc happens

I think that if we describe the base case here, it makes it easier to layer things on top of it (i.e. merchant) and helps with understanding without tying this to a specific p2p case only